fuse_models/src/unicycle_2d_state_kinematic_constraint.cpp

```diff
@@ -103,8 +102,16 @@ void Unicycle2DStateKinematicConstraint::print(std::ostream& stream) const
 
 ceres::CostFunction* Unicycle2DStateKinematicConstraint::costFunction() const
 {
-  return new ceres::AutoDiffCostFunction<Unicycle2DStateCostFunctor, 8, 2, 1, 2, 1, 2, 2, 1, 2, 1, 2>(
-    new Unicycle2DStateCostFunctor(dt_, sqrt_information_));
+  // Here we return a cost function that computes the analytic derivatives/jacobians, but we could use automatic
```

Comment:
I don't know if there's an easy way to allow to switch from analytic to automatic differentiation at runtime, but I guess we don't really need that. I might be convenient though for implementing the unit test that compares the jacobians with one and the other differentiation method, for validation purpose of the analytic ones.

Reply:
I'm somewhat curious to see the performance difference between the analytic derivatives and the automatic derivatives. As long as the analytic derivatives are faster, then I see no particular need to switch back and forth.